backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/connection/StorageServerConnectionCommandBase.java

```diff
@@ -80,10 +80,7 @@ protected String isConnWithSameDetailsExists(StorageServerConnections connection
             String connectionField = connection.getConnection();
             connections = storageServerConnectionDao.getAllForStorage(connectionField);
         } else {
-            StorageServerConnections sameConnection = iscsiStorageHelper.findConnectionWithSameDetails(connection);
-            connections =
-                    sameConnection != null ? Collections.singletonList(sameConnection)
-                            : Collections.emptyList();
+            connections = iscsiStorageHelper.findConnectionByAddressPortAndIqn(connection);
         }
 
         return (connections != null && connections.size() > 0 &&
```

Comment:
Sounds like `CollectionUtils.isNotEmpty(connections)`, but it's not your code anyway :)